src/python/grpcio_tests/tests/unit/_dynamic_stubs_test.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2019 The gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Test of dynamic stub import API."""
+
+import inspect
+import unittest
+import logging
+import contextlib
+import sys
+import multiprocessing
+import functools
+
+
+@contextlib.contextmanager
+def _protobuf_unimportable():
+    original_sys_path = sys.path
+    sys.path = [path for path in sys.path if "protobuf" not in path]
+    try:
+        yield
+    finally:
+        sys.path = original_sys_path
+
+
+def _wrap_in_subprocess(error_queue, fn):
+
+    @functools.wraps(fn)
+    def _wrapped():
+        try:
+            fn()
+        except Exception as e:
+            error_queue.put(e)
+            raise
+
+    return _wrapped
+
+
+def _run_in_subprocess(test_case):
+    error_queue = multiprocessing.Queue()
+    wrapped_case = _wrap_in_subprocess(error_queue, test_case)
+    proc = multiprocessing.Process(target=wrapped_case)
+    proc.start()
+    proc.join()
+    if not error_queue.empty():
+        raise error_queue.get()
+    assert proc.exitcode == 0, "Process exited with code {}".format(
+        proc.exitcode)
+
+
+def _assert_unimplemented(msg_substr):
+    import grpc
```

Comment:
Maybe we need a short comment to explain why we import grpc here, instead of importing at the top? I guess it is avoiding initialization for test case `test_protobuf_unimportable`.

Reply:
Yes. the intent here is only to do imports in subprocesses. Otherwise the test cases aren't actually independent.